Sming/Arch/Esp8266/Components/libc/component.mk

```diff
@@ -0,0 +1,13 @@
+COMPONENT_SRCDIRS		:=
+COMPONENT_INCDIRS		:= include
+COMPONENT_SRCFILES		:= libc.c
+
+ifeq ($(USE_NEWLIB),1)
+COMPONENT_SRCFILES		+= libc_replacements.c
+LIBDIRS					+= $(COMPONENT_PATH)/newlib
+EXTRA_LIBS				+= m c gcc
+else
+COMPONENT_SRCFILES		+= pgmspace.c
+LIBDIRS					+= $(COMPONENT_PATH)/lib
+EXTRA_LIBS				+= microc microgcc setjmp
+endif
```

Comment:
You may have spotted that `stdc++` has been removed from the list of libraries. One reason is that it contains allocation operators (new, delete, etc.) which conflict with our own. We don't use STL streams or a bunch of other stuff that it might be required for so leaving it out simplifies things. We can always revisit in the future if it's required for any other reason.

Reply:
Hello! I think that stdc++ is worth to keep. If there is any chance to do so. I use std::vector in my code and plan to switch to std:array all plain array. STL itself is good thing if used properly. Also there are libs that depend on STL flatbuffers as example. So do please keep this in Sming if possible.

Reply:
The STL shouldn't need the library as it's all templated code. It was only added in #1734 because of an issue with placeholders when compiling without optimisations.

It's probable that at least some parts of `libstdc++.a` will be required to support exception handling, RTTI, iostream, etc. but at present Sming doesn't use those.

Would you try some test compiles with your code to see if it builds OK without the library? Just remove the `stdc++` entry from `Arch/Esp8266/app.mk`. Post back it turns out there is stuff you do need from there then we can address that.

Reply:
Note: Host code gets build in GDB mode at `-O0` but as we've since updated Windows from GCC 5 to 8.2 it is no longer an issue.